File: app/src/main/java/com/example/movieapp102/repository/MovieRepository.java
package com.example.movieapp102.repository;

import androidx.lifecycle.LiveData;

import com.example.movieapp102.model.Movie;
import com.example.movieapp102.request.MovieApiClient;

import java.util.List;

public class MovieRepository {

    private static MovieRepository instance;

    private final MovieApiClient movieApiClient;


    public static MovieRepository getInstance() {
        if (instance == null) {
            instance = new MovieRepository();

        }
        return instance;
    }

    private MovieRepository() {
        movieApiClient = MovieApiClient.getInstance();
    }

    public LiveData<List<Movie>> getMovies() {
        return movieApiClient.getMovies();
    }

    public void searchMovieApi(String query,int page) { movieApiClient.searchMovieApi(query,page);}

}

File: app/src/main/java/com/example/movieapp102/repository/MovieRepository.kt
package com.example.movieapp102.repository

import androidx.lifecycle.LiveData
import com.example.movieapp102.model.Movie
import com.example.movieapp102.request.MovieApi

object MovieRepository {

	private var instance: MovieRepository = getInstance()

	private val movieApi: MovieApi = MovieApi.getInstance()

	fun getInstance(): MovieRepository {
		return instance
	}

	fun getMovies(): LiveData<List<Movie>> {
		return movieApi.getMovies()
	}

	fun searchMovie(query : String){
		movieApi.searchMovieApi(query)
	}

}